### PR TITLE
Display tooltip nearest cursor on non-bar chart

### LINKED ISSUE #2391: Hover over entire chart to show tooltip
* **C3 version**: 0.5.4
* **D3 version**: 5.2.0
* **Browser**: FF
* **OS**: MacOS

I think the [Google Group is dead](https://groups.google.com/forum/#!topic/c3js/VC2hKY44h1Q). Right?

I have an [older c3 instance][1] where you can hover over the chart and the tooltip is showing no matter where you point at on the chart.

In a [newer instance][2] this behaviour is gone.

Now the tooltip is only shown when hitting the datapoint excately.

Is there a option for this?

  [1]: https://aare.guru/aare-statistiken
  [2]: http://c3js.org/samples/timeseries.html

### PR DESCRIPTION
**Pull request by ajb32x**
I want to resolve ~~2362~~ 2391, this displays the tooltip for the closest point by horizontal distance to the cursor.
fixes #2391

### CODE REVIEW

**Comment by kt3k** [comment]
The preview example charts are available [here](https://540-11496279-gh.circle-artifacts.com/0/htdocs/index.html), but it doesn't seem fixing the problem reported in #2362.

File: src/data.js [hunk 1]
```diff
@@ -325,9 +325,17 @@ ChartInternal.prototype.findClosestFromTargets = function (targets, pos) {
...
+        config = $$.config,
+        xIndex = config.axis_rotated ? 1 : 0,
+        x = $$.x(data.x);
+
+        return Math.abs(x - pos[xIndex]);
+};
 ChartInternal.prototype.findClosest = function (values, pos) {
     var $$ = this,
-        minDist = $$.config.point_sensitivity,
+        minDist,
```
Comment (danidin) [inline src/data.js]: Seems like `minDist` is never defined

File: src/data.js [hunk 2]
```diff
@@ -344,13 +352,12 @@ ChartInternal.prototype.findClosest = function (values, pos) {
     values.filter(function (v) {
         return v && !$$.isBarType(v.id);
     }).forEach(function (v) {
-        var d = $$.dist(v, pos);
-        if (d < minDist) {
+        var d = $$.horizontalDistance(v, pos);
+        if (d < minDist || !minDist) {
```
Comment (Jupiterrr) [inline src/data.js]: You could initialize minDist with `Infinity` instead.

**Comment by ifokeev** [comment]
merge please

**Comment by 9teen90nine** [comment]
need this merged too!

**Comment by arnage** [comment]
great! merge it please

**Comment by adamkasprowicz** [comment]
:D btw if you can't wait for the merge - change this part of the code manually in node_modules. Works fine 👍 *unless you're building your bundle in CI pipeline

**Comment by ifokeev** [comment]
I found that PR doesn't resolve tooltip showing correctly. 
I noticed point_sensitivity config variable that isn't documented. 
You have to set 
```
        point: {
          sensitivity: 150
        },
```

and it will be actually the same behaviour as in the previous versions

**Comment by ajb32x** [comment]
> I found that PR doesn't resolve tooltip showing correctly.
> I noticed point_sensitivity config variable that isn't documented.
> You have to set
> 
> ```
>         point: {
>           sensitivity: 150
>         },
> ```
> and it will be actually the same behaviour as in the previous versions

This PR should fix the issue in #2391 , which looks to have been mistakenly marked as a duplicate of #2362 . I'm happy to make changes, I would like issue #2391 resolved.

My changes actually remove the use of the point_sensitivity variable for non bar charts. I don't believe it is necessary to fix #2391 .

**Comment by vlarkovich** [comment]
> The preview example charts (which includes this branch's change) are available [here](https://540-11496279-gh.circle-artifacts.com/0/htdocs/index.html), but it doesn't seem fixing the problem reported in #2362.

It seems that this PR will break the functionality of the tooltip when it is displayed separately for each point.
Try to open [not grouped tooltip](https://540-11496279-gh.circle-artifacts.com/0/htdocs/samples/tooltip_grouped.html), [scatter](https://540-11496279-gh.circle-artifacts.com/0/htdocs/samples/chart_scatter.html).

**Comment by kt3k** [comment]
Maybe we should have both `findClosest` and `findClosestHorizontally` and use one of them depending on the options specified.

**Comment by hln** [comment]
@vlarkovich is correct. When tooltip.grouped is set to false, the tooltip no longer works without having to directly hover over the points.

**Comment by kt3k** [comment]
Closing in favor of #2582.

@ajb32x Thank you for tackling this.

**Pull request closed without merging**